htlcswitch/switch.go

```diff
@@ -821,6 +834,10 @@ func (s *Switch) handleLocalDispatch(pkt *htlcPacket) error {
 		payment.preimage <- htlc.PaymentPreimage
 		s.removePendingPayment(pkt.incomingHTLCID)
 
+		if err := s.control.Success(pkt.circuit.PaymentHash); err != nil {
```

Comment:
In this case, and the one below, I think we want the control flow to be more like:
```golang
if err := s.control.Success(pkt.circuit.PaymentHash); err != nil {
    return err
}

payment := s.findPendingPayment(pkt.incomingHTLCID)
if payment != nil {
    payment.response <- pkt
    payment.preimage <- htlc.PaymentPreimage
    payment.preimage <- htlc.PaymentPreimage
    s.removePendingPayment(pkt.incomingHTLCID)
}
```

Currently we will return an error if we don't find the pending payment at the top of this method, which shouldn't actually be treated as an error (and would cause us not to write the state change at all). A pending payment won't exist after a restart, so we want to ensure we make the db transition before notifiying the application state.

I think because of this, we can safely change the signature of `findPendingPayment` to not return an error as well